airbyte-ci/connectors/metadata_service/lib/tests/test_gcs_upload.py

```diff
@@ -155,3 +156,41 @@ def test_upload_metadata_to_gcs_invalid_docker_images(mocker, invalid_metadata_u
                 "my_bucket",
                 metadata_file_path,
             )
+
+
+def test_upload_metadata_to_gcs_with_prerelease(mocker, valid_metadata_upload_files):
+    # Arrange
+    setup_upload_mocks(mocker, "new_md5_hash1", "new_md5_hash2", "new_md5_hash3")
+    mocker.patch("metadata_service.gcs_upload._latest_upload", return_value=None)
+    mocker.patch("metadata_service.gcs_upload.upload_file_if_changed", return_value=None)
+    mocker.spy(gcs_upload, "_prerelease_upload")
+
+    for valid_metadata_upload_file in valid_metadata_upload_files:
+        # Assuming there is a valid metadata file in the list, if not, you might need to create one
+        metadata_file_path = pathlib.Path(valid_metadata_upload_file)
+        prerelease_image_tag = "prerelease_image_tag"
+
+        gcs_upload.upload_metadata_to_gcs(
+            "my_bucket",
+            metadata_file_path,
+            prerelease_image_tag,
+        )
+
+        gcs_upload._latest_upload.assert_not_called()
+
+        # Assert that _prerelease_upload is called and the third argument is prerelease_image_tag
+        # Ignore the first and second arguments (_ and __ are often used as placeholder variable names in Python when you don't care about the value)
+        _, __, prerelease_tag = gcs_upload._prerelease_upload.call_args[0]
+        assert prerelease_tag == prerelease_image_tag
```

Comment:
Shall we add a similar "we called the right private method" assertion to the call to main upload [here](https://github.com/airbytehq/airbyte/pull/28067/files?diff=split&w=0#diff-8d2e2dfdaa33aa34583293472ad061493cf88a0ac2955e8bfe0cdc7480de10d1R100)?